docs/upgrading-pega-deployment-zero-downtime.md

```diff
@@ -0,0 +1,179 @@
+
+3. Runs the migration script, which creates any required tables in the new rules and temporary data schemas, and copies and moves the rule content.
+
+   ![Migrates the rules structure and content and data structure](media/upgrade-rules-rnew-schema.png "Migrates the rules structure and content and data structure")
+
+4. Upgrades the new rules schema. This process uses the temporary data schema in which it creates and modifies any necessary data tables to store interim data content; the process discards these tables and data after the upgrade completes.
+
+5. For deployments running in Oracle, grants access to the new rules schema tables to generate the necessary rules objects (GRANT statements) which will link the existing data schema to the new rules schema after you upgrade the new rules schema.
+
+6. Upgrades the existing data schema.
+
+7. Redeploys the application to run using the new rules and then performs a rolling restart of your deployment cluster.
+
+8. Reverts the high availability cluster settings to re-enable rule creation so your development team can create and extend rule creation.  
+
+Pega software upgrades do not touch the infrastructure of your Kubernetes environment.
+
+## Client-required steps
+
+Client-managed cloud clients use the Pega Kubernetes tools and Helm charts in the same Pega repository that you used to install Pega Platform in a supported Kubernetes environment. To complete an upgrade using these tools, you must perform the following tasks:
+
+1. Prepare your Docker repository by downloading the latest Pega-provided release images (platform/installer, platform/pega, and either platform/search or platform-services/search-n-reporting-service) in your release stream and pushing them into your preferred Docker image repository. For step-by-step details, see [Downloading and managing Pega Platform docker images (linux)](prepping-local-system-runbook-linux.md#downloading-and-managing-pega-platform-docker-images) or [Downloading and managing Pega Platform docker images (windows)](prepping-local-system-runbook-windows.md#downloading-and-managing-pega-platform-docker-images).
+
+2. Create a new blank rules schema and a temporary data schema in your existing database. Leave these new schemas empty. Note the names you use, since you must specify each in your upgrade configuration file.
+
+3. Edit the pega Helm chart as directed in [Upgrading Pega Platform with zero-downtime using Helm charts – 120 minutes](#upgrading-pega-platform-with-zero-downtime-using-helm-charts--120-minutes).
+
+4. Invoke the upgrade process by using the `helm upgrade release --namespace mypega` command as directed in the deployment section - [Upgrading your Pega Platform deployment using the command line](#upgrading-your-pega-platform-deployment-using-the-command-line).
+
+5. Delete the temporary data schema from your database after you complete your upgrade.
+
+Upgrading Pega Platform with zero-downtime using Helm charts – 120 minutes
```

Comment:
we cant say 120 mins its appx timing. Mostly will be in 2 hrs as we tried internally.